### PR TITLE
chore(remix-dev): mute spurious warnings when using Yarn

### PR DESCRIPTION
**Pull request by lensbart**
When running `yarn dev` in _some_ setups (e.g. in my case, when using Deno Deploy, but (if memory serves me well) not when using Remix  App Server), the following warnings are shown right after `Building Remix app in development mode...`:

```
The path "@remix-run/deno" is imported in server.ts but @remix-run/deno is not listed in your package.json dependencies. Did you forget to install it?
The path "@remix-run/react" is imported in src/entry.server.tsx but @remix-run/react is not listed in your package.json dependencies. Did you forget to install it?
The path "react" is imported in src/routes/index.tsx but react is not listed in your package.json dependencies. Did you forget to install it?
The path "react-dom/server" is imported in src/entry.server.tsx but react-dom is not listed in your package.json dependencies. Did you forget to install it?
```

These warnings are triggered because several packages don’t list their dependencies correctly according to (the somewhat more stringent) requirements of Yarn. This leads to several similar errors along the lines of “X tried to access Y, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound”. These errors are caught and the warnings above are shown instead.

I have been [informed](https://github.com/remix-run/remix/pull/3566#discussion_r906415009) that it is currently not desirable to update all package dependencies according to Yarn guidelines, in which case I think it’s best to mute these warnings for Yarn PnP users.

### CODE REVIEW

**Comment by MichaelDeBoey** [comment]
Since these are only warnings & don't block anything else, I think we can just leave this as-is.

As said in https://github.com/remix-run/remix/pull/3566#discussion_r906420249: the warning will disappear automatically once we made the switch to `@remix-run/router`.

**Comment by lensbart** [comment]
As I understood it, the migration to `@remix-run/router` will fix warnings on `react` and `react-dom` dependencies. The warnings above include other dependencies (e.g. `@remix-run/deno`, but presumably also for other integrations). Will the switch to the new router fix those as well?

In addition, the messaging of these warnings (“X is not listed in your package.json dependencies”) is incorrect.

**Comment by lensbart** [comment]
Hi @pcattori @mcansh, is there anything I can do to help this move along?

My console output for `yarn dev`:

```
The path "@remix-run/react" is imported in src/root.tsx but "@remix-run/react" was not found in your node_modules. Did you forget to install it?
The path "@apollo/client" is imported in src/routes/admin/organizations/controls/AddUserToOrganization.tsx but "@apollo/client" was not found in your node_modules. Did you forget to install it?
The path "react-intl" is imported in src/routes/profile/displayPreferences/DisplayPreferences.tsx but "react-intl" was not found in your node_modules. Did you forget to install it?
The path "react" is imported in src/routes/profile/displayPreferences/Card.tsx but "react" was not found in your node_modules. Did you forget to install it?
The path "react/jsx-dev-runtime" is imported in src/routes/admin/ui/Suggestions.tsx but "react/jsx-dev-runtime" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/react-fontawesome" is imported in src/routes/authors/CorrectionSection.tsx but "@fortawesome/react-fontawesome" was not found in your node_modules. Did you forget to install it?
The path "@remix-run/node" is imported in src/entry.server.tsx but "@remix-run/node" was not found in your node_modules. Did you forget to install it?
The path "deepmerge-ts" is imported in src/routes/admin/groups/controls/GrantGroupAdminPrivileges.tsx but "deepmerge-ts" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/free-solid-svg-icons/faLink" is imported in src/routes/team/Person.tsx but "@fortawesome/free-solid-svg-icons/faLink" was not found in your node_modules. Did you forget to install it?
The path "slate" is imported in src/routes/authors/editorSection/EditorSection.tsx but "slate" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faChartLine" is imported in src/routes/educational-institutions.tsx but "@private/fontawesome/pro-light-svg-icons/faChartLine" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faGrinWink" is imported in src/routes/publishers/RegistrationMarks.tsx but "@private/fontawesome/pro-light-svg-icons/faGrinWink" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faCheck" is imported in src/routes/authors/CorrectionSection.tsx but "@private/fontawesome/pro-light-svg-icons/faCheck" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/free-solid-svg-icons" is imported in src/routes/book/coverPage/CoverPage.tsx but "@fortawesome/free-solid-svg-icons" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons" is imported in src/routes/login/LoginForm.tsx but "@private/fontawesome/pro-regular-svg-icons" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faEnvelope" is imported in src/routes/publishers/index.tsx but "@private/fontawesome/pro-light-svg-icons/faEnvelope" was not found in your node_modules. Did you forget to install it?
The path "react-scroll" is imported in src/routes/privacy-policy.tsx but "react-scroll" was not found in your node_modules. Did you forget to install it?
The path "slate-react" is imported in src/routes/authors/editorSection/EditorSection.tsx but "slate-react" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faTable" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-light-svg-icons/faTable" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faPencilAlt" is imported in src/routes/authors/CorrectionSection.tsx but "@private/fontawesome/pro-light-svg-icons/faPencilAlt" was not found in your node_modules. Did you forget to install it?
The path "react-dom/server" is imported in src/entry.server.tsx but "react-dom/server" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faLink" is imported in src/routes/educational-institutions.tsx but "@private/fontawesome/pro-light-svg-icons/faLink" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faSmileWink" is imported in src/routes/publishers/RegistrationMarks.tsx but "@private/fontawesome/pro-light-svg-icons/faSmileWink" was not found in your node_modules. Did you forget to install it?
The path "js-cookie" is imported in src/routes/login/LoginForm.tsx but "js-cookie" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faEnvelopeOpen" is imported in src/routes/publishers/index.tsx but "@private/fontawesome/pro-light-svg-icons/faEnvelopeOpen" was not found in your node_modules. Did you forget to install it?
The path "@headlessui/react" is imported in src/routes/profile/index.tsx but "@headlessui/react" was not found in your node_modules. Did you forget to install it?
The path "react-router-dom" is imported in src/routes/admin/groups/AdminGroupPanel.tsx but "react-router-dom" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faLongArrowRight" is imported in src/routes/qr/index.tsx but "@private/fontawesome/pro-light-svg-icons/faLongArrowRight" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons" is imported in src/routes/library.tsx but "@private/fontawesome/pro-solid-svg-icons" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faBold" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-regular-svg-icons/faBold" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faTimes" is imported in src/routes/authors/CorrectionSection.tsx but "@private/fontawesome/pro-light-svg-icons/faTimes" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faStopwatch" is imported in src/routes/educational-institutions.tsx but "@private/fontawesome/pro-light-svg-icons/faStopwatch" was not found in your node_modules. Did you forget to install it?
The path "react-use" is imported in src/routes/profile/displayPreferences/Color.tsx but "react-use" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faPlusSquare" is imported in src/routes/profile/settings.tsx but "@private/fontawesome/pro-light-svg-icons/faPlusSquare" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons/faPlus" is imported in src/routes/library.tsx but "@private/fontawesome/pro-solid-svg-icons/faPlus" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faCode" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-regular-svg-icons/faCode" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faTrophyAlt" is imported in src/routes/educational-institutions.tsx but "@private/fontawesome/pro-light-svg-icons/faTrophyAlt" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faLongArrowRight" is imported in src/routes/book/coverPage/CoverPage.tsx but "@private/fontawesome/pro-regular-svg-icons/faLongArrowRight" was not found in your node_modules. Did you forget to install it?
The path "axios" is imported in src/routes/publish/index.tsx but "axios" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faFilm" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-regular-svg-icons/faFilm" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faImage" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-regular-svg-icons/faImage" was not found in your node_modules. Did you forget to install it?
The path "react-helmet-async" is imported in src/routes/book/coverPage/CoverPage.tsx but "react-helmet-async" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faIndent" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-regular-svg-icons/faIndent" was not found in your node_modules. Did you forget to install it?
The path "react-tsparticles" is imported in src/routes/educational-institutions.tsx but "react-tsparticles" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faArrowCircleDown" is imported in src/ui/Arrow.tsx but "@private/fontawesome/pro-light-svg-icons/faArrowCircleDown" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faItalic" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-regular-svg-icons/faItalic" was not found in your node_modules. Did you forget to install it?
The path "tsparticles" is imported in src/routes/educational-institutions.tsx but "tsparticles" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faLink" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-regular-svg-icons/faLink" was not found in your node_modules. Did you forget to install it?
The path "react-router" is imported in src/routes/publish/index.tsx but "react-router" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faListOl" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-regular-svg-icons/faListOl" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faListUl" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-regular-svg-icons/faListUl" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faOutdent" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-regular-svg-icons/faOutdent" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faRedo" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-regular-svg-icons/faRedo" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faUnderline" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-regular-svg-icons/faUnderline" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faUndo" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-regular-svg-icons/faUndo" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faArrowCircleRight" is imported in src/ui/landingPage/LibraryPreview.tsx but "@private/fontawesome/pro-light-svg-icons/faArrowCircleRight" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faQrcode" is imported in src/ui/form/QrInput.tsx but "@private/fontawesome/pro-regular-svg-icons/faQrcode" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons/faCaretDown" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-solid-svg-icons/faCaretDown" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons/faQuoteLeft" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@private/fontawesome/pro-solid-svg-icons/faQuoteLeft" was not found in your node_modules. Did you forget to install it?
The path "@react-spring/web" is imported in src/routes/authors/editorSection/Toolbar.tsx but "@react-spring/web" was not found in your node_modules. Did you forget to install it?
The path "react-dom" is imported in src/routes/authors/editorSection/Toolbar.tsx but "react-dom" was not found in your node_modules. Did you forget to install it?
The path "zxcvbn" is imported in src/ui/form/validatedInput/ValidatedInput.tsx but "zxcvbn" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons/faHeart" is imported in src/ui/landingPage/mockup/AnimatedComment.tsx but "@private/fontawesome/pro-solid-svg-icons/faHeart" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faSpinnerThird" is imported in src/ui/Spinner.tsx but "@private/fontawesome/pro-light-svg-icons/faSpinnerThird" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faEnvelope" is imported in src/ui/Link.tsx but "@private/fontawesome/pro-regular-svg-icons/faEnvelope" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/free-solid-svg-icons/faTimesCircle" is imported in src/ui/form/RemoveButton.tsx but "@fortawesome/free-solid-svg-icons/faTimesCircle" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/free-solid-svg-icons/faPenFancy" is imported in src/ui/Footer.tsx but "@fortawesome/free-solid-svg-icons/faPenFancy" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faExternalLink" is imported in src/ui/Link.tsx but "@private/fontawesome/pro-regular-svg-icons/faExternalLink" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/free-solid-svg-icons/faTachometerAlt" is imported in src/ui/Footer.tsx but "@fortawesome/free-solid-svg-icons/faTachometerAlt" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons/faBooks" is imported in src/ui/Footer.tsx but "@private/fontawesome/pro-solid-svg-icons/faBooks" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faAngleRight" is imported in src/content/annotation/discussion/Discussion.tsx but "@private/fontawesome/pro-regular-svg-icons/faAngleRight" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faTrashAlt" is imported in src/content/annotation/note/Note.tsx but "@private/fontawesome/pro-regular-svg-icons/faTrashAlt" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faWindowMinimize" is imported in src/content/annotation/note/Note.tsx but "@private/fontawesome/pro-regular-svg-icons/faWindowMinimize" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faChevronDown" is imported in src/ui/tableOfContents/RecursiveTree.tsx but "@private/fontawesome/pro-light-svg-icons/faChevronDown" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/free-solid-svg-icons/faBriefcase" is imported in src/ui/header/CompanyDropDownMenu.tsx but "@fortawesome/free-solid-svg-icons/faBriefcase" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons/faChevronDown" is imported in src/ui/tableOfContents/RecursiveTree.tsx but "@private/fontawesome/pro-solid-svg-icons/faChevronDown" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/free-solid-svg-icons/faUserCircle" is imported in src/ui/header/UserIcon.tsx but "@fortawesome/free-solid-svg-icons/faUserCircle" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/free-solid-svg-icons/faUserGraduate" is imported in src/ui/header/ProductDropDownMenu.tsx but "@fortawesome/free-solid-svg-icons/faUserGraduate" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faInfoCircle" is imported in src/ui/header/CompanyDropDownMenu.tsx but "@private/fontawesome/pro-light-svg-icons/faInfoCircle" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faSignIn" is imported in src/ui/header/UserIcon.tsx but "@private/fontawesome/pro-regular-svg-icons/faSignIn" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faBuilding" is imported in src/ui/header/ProductDropDownMenu.tsx but "@private/fontawesome/pro-light-svg-icons/faBuilding" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/free-solid-svg-icons/faFilter" is imported in src/ui/toolbar/buttons/Summary.tsx but "@fortawesome/free-solid-svg-icons/faFilter" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/free-solid-svg-icons/faAngleLeft" is imported in src/ui/toolbar/buttons/PreviousPage.tsx but "@fortawesome/free-solid-svg-icons/faAngleLeft" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/free-solid-svg-icons/faAngleRight" is imported in src/ui/toolbar/buttons/NextPage.tsx but "@fortawesome/free-solid-svg-icons/faAngleRight" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faUniversity" is imported in src/ui/header/ProductDropDownMenu.tsx but "@private/fontawesome/pro-light-svg-icons/faUniversity" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/free-solid-svg-icons/faList" is imported in src/ui/toolbar/buttons/TableOfContentsToggle.tsx but "@fortawesome/free-solid-svg-icons/faList" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faLongArrowDown" is imported in src/ui/guidedTour/steps/ScrollDown.tsx but "@private/fontawesome/pro-light-svg-icons/faLongArrowDown" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faTimes" is imported in src/ui/toolbar/buttons/Escape.tsx but "@private/fontawesome/pro-regular-svg-icons/faTimes" was not found in your node_modules. Did you forget to install it?
The path "@fortawesome/free-solid-svg-icons/faMoon" is imported in src/ui/toolbar/buttons/ColorTheme.tsx but "@fortawesome/free-solid-svg-icons/faMoon" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faMessage" is imported in src/ui/toolbar/annotate/CommentModeButton.tsx but "@private/fontawesome/pro-regular-svg-icons/faMessage" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-regular-svg-icons/faNote" is imported in src/ui/toolbar/annotate/NoteModeButton.tsx but "@private/fontawesome/pro-regular-svg-icons/faNote" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faFilter" is imported in src/ui/toolbar/buttons/Summary.tsx but "@private/fontawesome/pro-light-svg-icons/faFilter" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons/faMessage" is imported in src/ui/toolbar/annotate/CommentModeButton.tsx but "@private/fontawesome/pro-solid-svg-icons/faMessage" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons/faNote" is imported in src/ui/toolbar/annotate/NoteModeButton.tsx but "@private/fontawesome/pro-solid-svg-icons/faNote" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons/faLightbulb" is imported in src/ui/toolbar/buttons/ColorTheme.tsx but "@private/fontawesome/pro-solid-svg-icons/faLightbulb" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons/faLightbulbOn" is imported in src/ui/toolbar/buttons/ColorTheme.tsx but "@private/fontawesome/pro-solid-svg-icons/faLightbulbOn" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons/faCommentsAlt" is imported in src/ui/popup/selectionButtons/DiscussionButton.tsx but "@private/fontawesome/pro-solid-svg-icons/faCommentsAlt" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faDownload" is imported in src/content/block/file/File.tsx but "@private/fontawesome/pro-light-svg-icons/faDownload" was not found in your node_modules. Did you forget to install it?
The path "filesize" is imported in src/content/block/file/File.tsx but "filesize" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faFile" is imported in src/content/block/file/icon.ts but "@private/fontawesome/pro-light-svg-icons/faFile" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faFileAlt" is imported in src/content/block/file/icon.ts but "@private/fontawesome/pro-light-svg-icons/faFileAlt" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faFileArchive" is imported in src/content/block/file/icon.ts but "@private/fontawesome/pro-light-svg-icons/faFileArchive" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faFileAudio" is imported in src/content/block/file/icon.ts but "@private/fontawesome/pro-light-svg-icons/faFileAudio" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faFileChartPie" is imported in src/content/block/file/icon.ts but "@private/fontawesome/pro-light-svg-icons/faFileChartPie" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faFileCode" is imported in src/content/block/file/icon.ts but "@private/fontawesome/pro-light-svg-icons/faFileCode" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faFileCsv" is imported in src/content/block/file/icon.ts but "@private/fontawesome/pro-light-svg-icons/faFileCsv" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faFilePdf" is imported in src/content/block/file/icon.ts but "@private/fontawesome/pro-light-svg-icons/faFilePdf" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faFileSpreadsheet" is imported in src/content/block/file/icon.ts but "@private/fontawesome/pro-light-svg-icons/faFileSpreadsheet" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons/faPaperPlane" is imported in src/content/annotation/discussion/comment/controls/SendButton.tsx but "@private/fontawesome/pro-solid-svg-icons/faPaperPlane" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-solid-svg-icons/faUsers" is imported in src/content/annotation/discussion/comment/controls/VisibilitySelector.tsx but "@private/fontawesome/pro-solid-svg-icons/faUsers" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faFileVideo" is imported in src/content/block/file/icon.ts but "@private/fontawesome/pro-light-svg-icons/faFileVideo" was not found in your node_modules. Did you forget to install it?
The path "@private/fontawesome/pro-light-svg-icons/faFileWord" is imported in src/content/block/file/icon.ts but "@private/fontawesome/pro-light-svg-icons/faFileWord" was not found in your node_modules. Did you forget to install it?
```

**Review by mcansh — APPROVED** [review]

**Comment by pcattori** [comment]
@lensbart This is on my list of things to revisit for the week. I'll let you know if there's any way to make progress on this on our end.

**Comment by pcattori** [comment]
@lensbart from reading this and related issues, it seems like there are two causes for the spurious warnings:
1. Dependencies being pulled in by `react-router` that should be fixed when we internally switch to `@remix-run/router`
2. Other dependencies

It seems to me that for (2), it might be a good idea to do [what you suggested](https://github.com/remix-run/remix/pull/3566#discussion_r906418645):
```json
"peerDependenciesMeta": {
    "react": {
      "optional": true
    },
    "react-dom": {
      "optional": true
    }
  }
```

`peerDependenciesMeta` is part of the [package.json spec](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#peerdependenciesmeta), so its not yarn-specific, which is why I'm especially leaning towards that.

Beyond (1) being fixed via `@remix-run/router` soon, and fixing (2) via `peerDependenciesMeta` would there be anything else we would need to address?

**Comment by lensbart** [comment]
Thanks for looking into it, @pcattori!

> Beyond (1) being fixed via `@remix-run/router` soon, and fixing (2) via `peerDependenciesMeta` would there be anything else we would need to address?

Most of the warnings quoted above are not due to a missing `peerDependenciesMeta` setting. The check for this warning (which this PR circumvents) fails to detect some dependencies that are indeed available. Not sure why, but doesn’t tooling (ESLint / TypeScript) typically help warn against uninstalled dependencies, anyway?

The `@private/...` dependencies you see listed are installed from a private repository. Yarn can be configured to install from a private repository as follows:

```yml
# .yarnrc.yml
npmScopes:
  private:
    npmAlwaysAuth: true
    # Include trailing `-` to prevent `yarn` from failing in CI environment where `AWS_CODEARTIFACT_AUTH_TOKEN` is not available.
    npmAuthToken: '${AWS_CODEARTIFACT_AUTH_TOKEN-}'
    npmRegistryServer: 'https://something.d.codeartifact.us-east-1.amazonaws.com/npm/private'
```

In conclusion, I think the check for the warning should either be compatible with all package managers and resolution mechanisms (npm, yarn, yarn pnp, pnpm), or be disabled for those that are incompatible.

Hope this helps!

**Comment by pcattori** [comment]
> In conclusion, I think the check for the warning should either be compatible with all package managers and resolution mechanisms (npm, yarn, yarn pnp, pnpm), or be disabled for those that are incompatible.

Well said; I agree.

File: packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts [hunk 1]
```diff
@@ -70,7 +70,10 @@ export function serverBareModulesPlugin(
         if (
           onWarning &&
           !isNodeBuiltIn(packageName) &&
-          !/\bnode_modules\b/.test(importer)
+          !/\bnode_modules\b/.test(importer) &&
+          // silence spurious warnings when using Yarn PnP until issues with
+          // ambiguous/unsound extraneous dependencies are fixed
+          process.versions.pnp == null
```
Comment (pcattori) [inline packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts:76]: If I understand this correctly, we're currently looking exclusively for Node built-ins or in `node_modules/` for dependencies, but Yarn PnP doesn't put things into `node_modules/`, so our existing check fails and thinks the dependency is missing.

@lensbart is that an accurate paraphrasing?

If so, could you include something to that effect in the comment here explaining _why_ checking `node_modules/` is insufficient for Yarn PnP?

Comment (pcattori) [reply packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts:76]: Also instead of _disabling_ warnings for Yarn PnP, it'd be better to _enable_ them for compatible package managers.

Comment (lensbart) [reply packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts]: @pcattori Updated the comment!

Comment (lensbart) [reply packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts]: Not sure how to check for other package managers such as pnpm, so I kept the check as-is.

**Comment by lensbart** [comment]
@pcattori Is there anything I can do to help move this forward? I also created https://github.com/remix-run/remix/issues/4346 to summarise other outstanding issues for Yarn PnP support.

Thank you!

File: packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts (hunk 1 above)
Comment (pcattori) [reply packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts]: @lensbart we have a [`getPreferredPackageManager` utility](https://github.com/remix-run/remix/blob/dev/packages/remix-dev/cli/getPreferredPackageManager.ts) that you can use to check for the package manager.

Could you use that to enable the warnings for just `npm` and `yarn`?

**Comment by pcattori** [comment]
@lensbart sorry for the delay! I was busy getting married last week 😅

If you can just change the warnings to be enabled for `npm` and `yarn` (rather than being disabled for `pnpm`), then I think we're good to merge this 😁

File: packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts (hunk 1 above)
Comment (lensbart) [reply packages/remix-dev/compiler/plugins/serverBareModulesPlugin.ts]: Consider it done! Except I’ve disabled the warnings for `yarn`, assuming that’s what you meant.

**Comment by lensbart** [comment]
Congrats on getting married, by the way!!! 🥳🥳🥳

**Comment by lensbart** [comment]
> If you can just change the warnings to be enabled for `npm` and `yarn` (rather than being disabled for `pnpm`)

Just for the avoidance of confusion: Plug‘n’Play [is a feature of Yarn](https://next.yarnpkg.com/features/pnp), abbreviated as PnP. This is unrelated to the `pnpm` package manager.

**Comment by pcattori** [comment]
Looking at this again, I just realized I was talking as if this PR was for pnpm support, not yarn pnp.

So what I'm actually looking for here is something like:

```ts
if (usingNode || usingBasicYarn) {
  warn()
}
```

Specifically, I think we still need the `process.versions.pnp` check to differentiate basic Yarn from Yarn PnP.

So maybe something like:

```ts
let pkgManager = getPreferredPackageManager()
if (pkgManager === "node" || (pkgManager === "yarn" && process.versions.pnp === null)) {
  warn()
}
```

**Comment by mcansh** [comment]
interestingly enough, I'm looking into upgrading esbuild to use the native pnp stuff over there and i don't get any warnings with it https://github.com/mcansh/remix-pnp-test

`0.0.0-experimental-aed5bfbf7`

**Comment by lensbart** [comment]
@pcattori Done!

**Comment by lensbart** [comment]
@mcansh Interesting. Does that also fix the issue with import aliases, described as the third problem in [this issue](https://github.com/remix-run/remix/issues/4346)?

**Comment by mcansh** [comment]
> @mcansh Interesting. Does that also fix the issue with import aliases, described as the third problem in [this issue](https://github.com/remix-run/remix/issues/4346)?

it seems to, but typescript (in vscode) is furious as it doesn't know about pnp 
image

**Comment by lensbart** [comment]
@mcansh Make sure to [`yarn dlx @yarnpkg/sdks vscode`](https://yarnpkg.com/getting-started/editor-sdks#editor-setup) for editor support

**Comment by mcansh** [comment]
> @mcansh Make sure to [`yarn dlx @yarnpkg/sdks vscode`](https://yarnpkg.com/getting-started/editor-sdks#editor-setup) for editor support

just did 😆 - i know very little about pnp haha - if you want to toy around `0.0.0-experimental-aed5bfbf7` is available for all remix packages 😄

**Comment by lensbart** [comment]
@mcansh I can get it to work with that experimental branch, thanks. (I still see the redundant warnings that this PR fixes.)

**Comment by mcansh** [comment]
@lensbart thanks for giving it a run! surprised i didn't get the warnings, so good thing we merged this 🙂
